src/main/java/org/springframework/data/redis/connection/jedis/JedisClusterConnection.java

```diff
@@ -887,9 +882,38 @@ public ClusterTopology getTopology() {
 		 *         topology.
 		 * @see #JedisClusterTopologyProvider(JedisCluster, Duration)
 		 * @since 2.2
+		 * @deprecated since 3.3.4, use {@link #shouldUseCachedValue(JedisClusterTopology)} instead.
 		 */
+		@Deprecated(since = "3.3.4")
 		protected boolean shouldUseCachedValue() {
-			return time + cacheTimeMs > System.currentTimeMillis();
+			return false;
+		}
+
+		/**
+		 * Returns whether {@link #getTopology()} should return the cached {@link JedisClusterTopology}. Uses a time-based
+		 * caching.
+		 *
+		 * @return {@literal true} to use the cached {@link ClusterTopology}; {@literal false} to fetch a new cluster
+		 *         topology.
+		 * @see #JedisClusterTopologyProvider(JedisCluster, Duration)
+		 * @since 3.3.4
+		 */
+		protected boolean shouldUseCachedValue(@Nullable JedisClusterTopology topology) {
+			return topology != null && topology.getTime() + cacheTimeMs > System.currentTimeMillis();
```

Comment:
we could move the `cacheTimeMs` to the `JedisClusterTopology` having a package private method `maxTime` that does the calculation.